plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -238,37 +274,60 @@ public static ReaderPageSource createPageSource(
             messageColumn = getColumnIO(fileSchema, requestedSchema);
 
             Map<List<String>, ColumnDescriptor> descriptorsByPath = getDescriptors(fileSchema, requestedSchema);
-            TupleDomain<ColumnDescriptor> parquetTupleDomain = options.isIgnoreStatistics()
-                    ? TupleDomain.all()
-                    : getParquetTupleDomain(descriptorsByPath, effectivePredicate, fileSchema, useColumnNames);
-
-            TupleDomainParquetPredicate parquetPredicate = buildPredicate(requestedSchema, parquetTupleDomain, descriptorsByPath, timeZone);
+            List<ParquetDomainAndPredicate> parquetDomains = new ArrayList<>();
+            FilterPredicate nullableFilterPredicate = null;
+            for (TupleDomain<HiveColumnHandle> domain : effectivePredicates) {
```

Comment:
`options.isIgnoreStatistics()` can be consulted once.
it never changes. would make the code simpler to follow.

Reply:
```java
List<ParquetDomainAndPredicate> domainAndPredicates;
            Optional<FilterPredicate> filterPredicate;
            if (!options.isIgnoreStatistics()) {
                domainAndPredicates = effectivePredicates.stream()
                        .map(domain -> {
                            TupleDomain<ColumnDescriptor> parquetTupleDomain = getParquetTupleDomain(descriptorsByPath, domain, fileSchema, useColumnNames);
                            TupleDomainParquetPredicate parquetPredicate = buildPredicate(requestedSchema, parquetTupleDomain, descriptorsByPath, timeZone);
                            return new ParquetDomainAndPredicate(parquetTupleDomain, parquetPredicate);
                        })
                        .collect(toImmutableList());

                filterPredicate = domainAndPredicates.stream()
                        .map(ParquetDomainAndPredicate::predicate)
                        .map(parquetPredicate -> parquetPredicate.toParquetFilter(timeZone))
                        .flatMap(Optional::stream)
                        // TODO create balanced tree when this is used with high number of filters. Currently it's used with at most 2
                        .reduce(FilterApi::or);
            }
            else {
                domainAndPredicates = ImmutableList.of();
                filterPredicate = Optional.empty();
            }
```